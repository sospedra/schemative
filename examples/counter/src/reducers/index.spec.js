import counter from './index'
import defaultState from '../state'

describe('reducers', () => {
  describe('counter', () => {
    const incrementedState = Object.assign({
      value: 1
    }, defaultState)

    it('should provide the initial state', () => {
      expect(counter(undefined, {})).toBe(defaultState)
    })

    it('should handle INCREMENT action', () => {
      expect(counter(incrementedState, { type: 'INCREMENT' })).toMatchSnapshot()
    })

    it('should handle DECREMENT action', () => {
      expect(counter(incrementedState, { type: 'DECREMENT' })).toMatchSnapshot()
    })

    it('should ignore unknown actions', () => {
      expect(counter(incrementedState, { type: 'unknown' })).toMatchSnapshot()
    })
  })
})
