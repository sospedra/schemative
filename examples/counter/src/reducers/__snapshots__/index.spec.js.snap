exports[`reducers counter should handle DECREMENT action 1`] = `
Object {
  "value": -2
}
`;

exports[`reducers counter should handle INCREMENT action 1`] = `
Object {
  "value": 0
}
`;

exports[`reducers counter should ignore unknown actions 1`] = `
Object {
  "value": -1
}
`;
